Bottom: 94fb50a42e9fcae41ff6481b57a6dfd6e8209617
Top:    9259926ad7ea4d460f8ec1f7aa77fe6499d7c4f7
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-12-04 15:55:49 +0000

Fix to ParseGroupString()


---

diff --git a/src/misc.c b/src/misc.c
index 040124c..545aaac 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -1136,7 +1136,14 @@ ParseGroupString (char *group_string, LayerGroupType *LayerGroup, int *LayerN)
           /* ignore white spaces and check for separator */
           while (*s && isspace ((int) *s))
             s++;
-          if (!*s || *s == ':')
+
+          /* Backup so the outer loop terminates on the NULL termination */
+          if (*s == '\0') {
+            s--;
+            break;
+          }
+
+          if (*s == ':')
             break;
         }
       LayerGroup->Number[group] = member;
